### ISSUE TITLE
Possible to register a directory for blocks that isnt app/Blocks?

### ISSUE THREAD

**Issue by drbroad** [Log1x/acf-composer#240]
In previous versions, I have been able to have my block .php files in a custom directory under an autoloaded namespace.
With the newest version, it doesn't pick up these block definitions and make them available in the editor

Can you do something like the config/views.php file where you can specify other directories to look in?

eg: 

`<?php

return [
    'paths' => [
        get_theme_file_path('/mypath/views'),
        get_parent_theme_file_path('/mypath/views'),
        get_theme_file_path('/resources/views'),
        get_parent_theme_file_path('/resources/views'),
    ],

];
`

**Comment by mike-sheppard** [comment]
@drbroad yep that's pretty much what we do to load block views in their own directory.


```php
// config/view.php
'paths' => [
    resource_path('views'),
    resource_path('blocks'), // our custom blocks view dir
],
```

```php
// app/Blocks/Hero.php
public $view = 'hero.hero';
```

Which will load in `resources/blocks/hero/hero.blade.php`

**Comment by drbroad** [comment]
@mike-sheppard Apologies, I just reread my post and I wasn't exactly clear.

I already have the view files registered as per your example, but I'm trying to get custom Block definitions registered. (Example.php for example) 

I have tried doing the following:

Assuming I have a structure of:
`
- CustomBlockFolder
-- Carousel
--- ...other assets for the carousel block
--- Blocks
---- Block.php
-- Slider
--- ...other assets for the slider block
--- Blocks
---- Block.php
`

I have used the following in older versions to iterate over that folder and register all the Block files:
`
        $app = new AcfComposer(app());
        $dirs = glob(get_theme_file_path('/customBlockPaths/*'), GLOB_ONLYDIR);

        foreach ($dirs as $d) {
            $name   = basename($d);
            $class  = "CustomBlockPaths\\" . $name . '\\';
            $p = $app->registerPath($d, $class);
        }

        var_dump($app->paths());

`
Again, this was code from 2.1.0 - but that method still exists in the latest release

I can see that it will add my block to the paths, but it still isn't viewable in the admin (I took the exact output of ```wp acorn acf:block Example``` and adjusted the namespace to match the folder structure)

**In case you are wondering the motivation behind the request**: I have all my block elements in separate repos that I can import into any of my projects. It is a self-contained folder with all the assets, scripts, block definition, and view - so that it can be dropped into any project and basically 'just work'.

**Comment by mike-sheppard** [comment]
Aha, yeah, that does make sense 👍 Unfortunately I won't be much help on this so I'll leave it for @Log1x or other folks to follow up on! :)

**Comment by drbroad** [comment]
If I hardcode the same values into `AcfComposer.php` on line 172 as I would be from my loop:

```
    /**
     * Register the default application path.
     */
    public function registerDefaultPath(): void
    {
        $this->registerPath($this->app->path());
        $this->registerPath('/Users/me/Desktop/www/acftest/site/web/app/themes/acftest/customBlockPaths/Carousel', 'CustomBlockPaths\Carousel\\');
);
    }
```

Everything works just fine.

it seems that when I am running the following, it isn't being registered at the same time / is being registered to a different instance of AcfComposer:
```
        $app = new AcfComposer(app());
        $dirs = glob(get_theme_file_path('/customBlockPaths/*'), GLOB_ONLYDIR);

        foreach ($dirs as $d) {
            $name   = basename($d);
            $class  = "CustomBlockPaths\\" . $name . '\\';
            $p = $app->registerPath($d, $class);
        }

```

I can test it by running `wp acorn acf:block Example` and then:
```
$app = new AcfComposer(app());
var_dump($app->paths());
```
Which returns an empty array. My thinking was this should at least return the default path and existing blocks.

Should I be calling registerPath() a different way?

**Comment by Log1x** [comment]
Try calling it from a service provider like:

`app('AcfComposer')->registerPath(...)`

But I admittedly need to test this stuff more. I think adding an option to load classes from the config would be a fine addition too.

**Comment by drbroad** [comment]
> `app('AcfComposer')->registerPath(...)`

Boy, do I feel silly right now - this works perfectly, thank you!
Also, a config approach would be very useful too.

**Comment by drbroad** [comment]
@Log1x this approach has been working great, up until Laravel Acorn 5.0.0-beta.2 (Laravel 11.34.2)
Now, this approach results in a target class not found:

```
    public static function registerPageBlocks($exclude = array())
    {
        $app = new AcfComposer(app());
        $dirs = glob(get_theme_file_path('/fir/Pinecones/*'), GLOB_ONLYDIR);

        foreach ($dirs as $d) {
            $name   = basename($d);
            if (in_array($name, $exclude)) {
                continue;
            }

            $class  = "Fir\\Pinecones\\" . $name . '\\';

            app('AcfComposer')->registerPath($d, $class);
        }
    }
```

Results in:
<img width="1467" alt="Screenshot 2025-01-08 at 9 59 49 AM" src="https://github.com/user-attachments/assets/76e458e3-a6f1-4ae8-b099-8aea428fd9dd" />

Am I doing something silly?

**Comment by Log1x** [comment]
hmm, perhaps try wrapping it in an `after_setup_theme` hook or something? otherwise, it should ideally work as-is in the `boot` method of a Service Provider.

**Comment by drbroad** [comment]
Nuked it, reran everything- worked fine.

¯\_(ツ)_/¯
